lib/guest_installation_and_configuration_base.pm

```diff
@@ -401,9 +403,12 @@ sub prepare_non_transactional_environment {
         virt_autotest::utils::setup_rsyslog_host($common_log_folder);
         my $_packages_to_check = 'wget curl screen dnsmasq xmlstarlet yast2-schema python3 nmap';
         zypper_call("install -y $_packages_to_check");
-        my $_patterns_to_check = 'kvm_server kvm_tools';
-        $_patterns_to_check = 'xen_server xen_tools' if ($self->{host_virt_type} eq 'xen');
-        zypper_call("install -y -t pattern $_patterns_to_check");
+        # There is already the highest version for kvm/xen packages on TW
+        if (is_sle) {
+            my $_patterns_to_check = 'kvm_server kvm_tools';
+            $_patterns_to_check = 'xen_server xen_tools' if ($self->{host_virt_type} eq 'xen');
+            zypper_call("install -y -t pattern $_patterns_to_check");
```

Comment:
```if （!is_tubleweed)``` better ?

Reply:
'is_tubleweed' does not follow the code convention about factory-first which was warned by Oliver a few times above. It should be used with limitation. Is 'is_sle' at  here correct? Are there any other products such as sle_micro or ALP required as well? If so, I have to reconsider of 'is_tumbleweed'.

Reply:
1.This function is prepare_non_transactional_environment, so only take effect on non-transactional ones. Maybe you can keep it like this at the moment, and add more checking later if necessary. 
2.If you just want to skip this part because there is already highest version of them, I think you do not need to do anything, just leave it as what it was before.

Reply:
> 2. If you just want to skip this part because there is already highest version of them, I think you do not need to do anything, just leave it as what it was before.

because zypper still do some fresh or querry work to connect remote repo, download.opensuse.org, which has risks of timeout or failures. I am trying to avoid any unnecessary connection to download.opensuse.org as its unstable problem seems not to be fixed in short time.